src/js/tabs/firmware_flasher.js

```diff
@@ -774,6 +772,75 @@ TABS.firmware_flasher.initialize = function (callback) {
             }
         }
 
+        function verifyBoard() {
+            if (!$('option:selected', portPickerElement).data().isDFU) {
+                let board = "UNKNOWN";
+
+                function onFinishClose() {
+                    MSP.clearListeners();
+                    GUI.connect_lock = false;
+                }
+
+                function onClose() {
+                    serial.disconnect(onFinishClose);
+                    MSP.disconnect_cleanup();
+                }
+
+                function onFinish() {
+                    board = FC.CONFIG.boardName;
+                    if (board) {
+                        self.verifiedBoard = board;
+                        $('select[name="board"]').val(board).trigger('change');
+                        GUI.log(i18n.getMessage('firmwareFlasherBoardVerificationSuccess', {boardName: board}));
+                    } else {
+                        GUI.log(i18n.getMessage('firmwareFlasherBoardVerificationFail'));
+                    }
+                    onClose();
+                }
+
+                function getBoard() {
+                    console.log(`Requesting board information`);
+                    MSP.send_message(MSPCodes.MSP_API_VERSION, false, false, () => MSP.send_message(MSPCodes.MSP_BOARD_INFO, false, false, onFinish));
```

Comment:
Not sure why we are sending `MSP_API_VERSION` here?
In general, I think it would be nice if we could have the MSP call stack for detecting the board version only once, and then use it here and in the check that is in `stm32.js` - this will make it easier to maintain.

Reply:
The value `FC.CONFIG.apiVersion` is needed in the `MSP_BOARD_INFO` call
Due to changing ports / having to have a connection (?) the values in `FC.CONFIG` becomes empty,  not sure if this has anything to do with the transition to modules or if this how it works.

The board is verified:

- on tab entry
- on port change
- when going to flash I present the dialog if verified board is not the selected board.

Reply:
The value `FC.CONFIG.apiVersion` is needed in the `MSP_BOARD_INFO` call
Due to changing ports / having to have a connection (?) the values in `FC.CONFIG` becomes empty,  not sure if this has anything to do with the transition to modules or if this how it works.